zhaquirks/tuya/ts0041.py

```diff
@@ -184,3 +184,56 @@ class TuyaSmartRemote0041TOPlusA(CustomDevice):
         (LONG_PRESS, BUTTON_1): {ENDPOINT_ID: 1, COMMAND: LONG_PRESS},
         (DOUBLE_PRESS, BUTTON_1): {ENDPOINT_ID: 1, COMMAND: DOUBLE_PRESS},
     }
+
+
+class TuyaSmartRemote0041_var04(CustomDevice):
+    """Tuya 1-button remote device with time on out cluster."""
+
+    signature = {
+        MODEL: "TS0041A",
+        ENDPOINTS: {
+            # "profile_id": 260,
+            # "device_type": "0x0000",
+            # "in_clusters": ["0x0000","0x0001","0x0006"],
+            # "out_clusters": ["0x0006","0x000a","0x0019"]
+            1: {
+                PROFILE_ID: zha.PROFILE_ID,
+                DEVICE_TYPE: zha.DeviceType.ON_OFF_SWITCH,
+                INPUT_CLUSTERS: [
+                    Basic.cluster_id,
+                    PowerConfiguration.cluster_id,
+                    OnOff.cluster_id,
+                ],
+                OUTPUT_CLUSTERS: [
+                    OnOff.cluster_id,
+                    Time.cluster_id,
+                    Ota.cluster_id,
+                ],
+            },
+        },
+    }
+
+    replacement = {
+        ENDPOINTS: {
+            1: {
+                PROFILE_ID: zha.PROFILE_ID,
+                DEVICE_TYPE: zha.DeviceType.ON_OFF_SWITCH,
+                INPUT_CLUSTERS: [
+                    Basic.cluster_id,
+                    PowerConfiguration.cluster_id,
+                    TuyaSmartRemoteOnOffCluster,
+                ],
+                OUTPUT_CLUSTERS: [
+                    TuyaSmartRemoteOnOffCluster,
```

Comment:
Are both clusters used by that remote?
Or is replacing one `OnOff` cluster enough?

Reply:
There is only one button, but i really dont know why it has one input and one output OnOff cluster.
It does not throw any errors or warnings as it is now, but We could try with replacing only one.

Reply:
Its one tuya thing and deleting it in replacement is making ZHA not making on switch in the GUI that is not working (at least for TS004F devices).
Testing if it working without it and its still sending commands OK and not getting problems / delays with fast key pressings.

Reply:
So when only one cluster is replaced, it doesn’t work?

Reply:
I'll have to do another test. Should I replace TuyaSmartRemoteOnOffCluster with OnOff.cluster_id in the output cluster?

Reply:
Try leaving it out completely, so you have just one Tuya OnOff cluster.
(If one doesn’t work, try the other one.)

Reply:
I tought that the number of output cluster must match the number in the device signature, otherwise it will fail loading the quirk. This is my proposed replacement:

`replacement = {
        ENDPOINTS: {
            1: {
                PROFILE_ID: zha.PROFILE_ID,
                DEVICE_TYPE: zha.DeviceType.ON_OFF_SWITCH,
                INPUT_CLUSTERS: [
                    Basic.cluster_id,
                    PowerConfiguration.cluster_id,
                    TuyaSmartRemoteOnOffCluster,
                ],
                OUTPUT_CLUSTERS: [
                    OnOff.cluster_id,,
                    Time.cluster_id,
                    Ota.cluster_id,
                ],
            },
        },
    }`

Reply:
The above code is the only one that is working reliably. PR has been updated with tuya cluster in only the input cluster.

Reply:
Ar you getting one switch in ZHA GUI then having it as input cluster after restarting HA ?
All TS004X is having  problems with the dealt  response shall being sent the wrong way or the device is making one pause between commands if making more fast. But it can being different  with the **A** version you is having.

I have making one issue for sending it in the wrong detection but its not solved yet.

Reply:
@MattWestb It does expose a switch entity in the GUI. I will check how it works. Might need some more work, but basic functionality of the button is in place.

Reply:
This code does not provide the switch entity in the GUI as previous versions did. 
If you have used a previous version, you will have to delete the switch entity manually as it is no longer provided by the integration, it will still be there after a restart if not deleted manually.